src/utils.cpp

```diff
@@ -334,6 +335,203 @@ void cnf_to_dimacs(Term cnf, std::ostringstream & y)
   }
 }
 
+//mapping each subformula with a new name and returning a vector of pair of terms. 
+//Each pair consists of the parent term and it's children(for each subformula)
+class traversal : public IdentityWalker
+{
+ public:
+  std::vector<std::pair<Term, Term>> reduced;
+  static int pt;  // the static variable to always produce distinct symbols
```

Comment:
@makaimann an idea we might have here is to expose a unique pointer to the `traversal` class. Would that solve this issue? And do you think this would be better than a static member?

Reply:
Right, so it seems like you would want a "static" member for each solver (e.g., if you had two `traversal` objects using the same solver). But otherwise, you don't want a static member because you can reuse symbol names. It's not the most performance efficient, but for a first pass, I think a non-static `pt` with a `get_unique_symbol` function would be fine. Something like this (didn't check if it would compile):

```
Term traversal::get_unique_symbol()
{
  while(true)
  {
     try
     {
        return solver_->make_symbol("fresh_symbol_" + to_string(pt), boolsort_);
     }
     catch(IncorrectUsageException & e)
     {
       ++pt;
     }
  }
}
```

I don't think you need to worry about the edge case (enumerating all possible pt values). But you could always put a condition in the while-loop for that if needed. Also, I'd prefer the class name to be capitalized: `Traversal`. And maybe a more descriptive name would be good as well.